ion/unmarshal.go

```diff
@@ -671,3 +701,48 @@ func indirect(v reflect.Value, wantPtr bool) reflect.Value {
 
 	return v
 }
+
+func (d *Decoder) decodeToStructWithAnnotation(v reflect.Value) error {
+	if err := d.attachAnnotations(v); err != nil {
+		return err
+	}
+
+	fields := fieldsFor(v.Type())
+	for _, field := range fields {
+		if field.name != AnnotationsTag {
```

Comment:
Trying to stash the value into the first field that isn't named "annotations" feels like it could be error-prone. Would it make sense to require the value-receiving field to be similarly annotated with something like `ion:",value"`?

Reply:
Agree, to some extent. Randomly grabbing the first field (except "annotation") to put the data in, for sure, is what I do not like about about my solution, but apart from not having a better idea:
- I did not want to impose another restriction on user, to tag the field for value as well as the one for annotations.
- I cannot think of a case where a struct need to have more than 2 fields (1 for annotations and the other for the value)

Reply:
Maybe leave the value field untagged but verify before taking this codepath that the struct has exactly two fields, one of type `[]string` tagged with `ion:"annotations"` and the other of which will receive the value?

Reply:
Updated the PR.
Thanks.